### DISCUSSION TITLE
AOTE is sending old MQTT-Topics

### DISCUSSION THREAD

**Discussion by Viko1690** [jomjol/AI-on-the-edge-device#3208] [Q&A]
Hey all,

first many thanks for the great work at this project!  After many try and error to get the values over mqtt/mosquitto and Telegraf into InfluxDBv2 (everything run as docker containers), I was able to get it work. But now there is one problem left.

During my tests i played with the mqtt settings of AOTE (enable/disable retain flag, HA discovery, change the main topic).

In my working configuration the retain flag and ha discovery is disabled but AOTE is still sending the ha topics (homeassistent/sensor/…). Also the older topics which I defined during my tests are sent too. 

I thought, that maybe some retained message still get still flushed to influx, so I deleted and redeployed the mosquitto and Telegraf containers but the problem still exist. The next weird thing… I do not have even subscribed the old topics and ha topics in my Telegraf configuration but they appeared in my influxdb.

I am using the firmware 15.7. Does anyone has an idea?

Thank you in advance!

Here is my Telegraf config for further information

```
[[inputs.mqtt_consumer]]
  servers = ["tcp://mosquitto:1883"]
  topics = [
    "meter/#"
  ]
  data_format = "value"
  data_type = "auto_float"

  [[inputs.mqtt_consumer.topic_parsing]]
    topic = "meter/+/+"
    tags = "_/source/field"
    [[processors.pivot]]
      tag_key = "field"
      value_key = "value"

  [[inputs.mqtt_consumer.topic_parsing]]
    topic = "meter/+/+/+"
    tags = "_/source/_/field"
    [[processors.pivot]]
      tag_key = "field"
      value_key = "value"

[[outputs.influxdb_v2]]
  urls = ["http://influxdb:8086"]
  token = `"xxx-one-very-secret-token-xxx"`
  organization = "Home"
  bucket = "EnergizerData" 
```

**Comment by caco3** [comment]
> ha discovery is disabled but AOTE is still sending the ha topics (homeassistent/sensor/…)

That would be wrong, indeed.

But old topics sound like issues outside of AIOTED. Because after changing the topic and saving it and restarting the device (you did, right?), the device has no more knowledge of the old topics.

please switch the logging to DEBUG and provide us a debug log. IIRC, each MQTT publishing gets noted as a debug log.

**Comment by Slider0007** [comment]
Most likely you only need to delete the old persistent HA topics at MQTT broker side.
